game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -317,10 +355,27 @@ public int getDistance(final Territory t1, final Territory t2) {
    * @param cond condition that covered territories of the route must match
    */
   public int getDistance(final Territory t1, final Territory t2, final Predicate<Territory> cond) {
+    return getDistance(t1, t2, cond, (it, it2) -> true);
+  }
+
+  /**
+   * Returns the distance between two territories where the covered territories of the route satisfy
+   * the condition or -1 if they are not connected.
+   *
+   * @param t1 start territory of the route
+   * @param t2 end territory of the route
+   * @param cond condition that covered territories of the route must match
+   * @param routeCond condition that routes between neighboring territories must match
+   */
+  public int getDistance(
```

Comment:
Similar comment to this method, it's either invoked with a no-op arg or the arg is always:
```
                  ProMatches.territoryCanPotentiallyMoveLandUnits(player, data.getProperties()),
                  ProMatches.noCanalsBetweenTerritories(player, data));
```

That hints that we could use another method rather than it need to be parameterized as part of the public API.